src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -759,19 +762,21 @@ Namespace Microsoft.CodeAnalysis.Operations
         End Function
 
         Private Function CreateBoundWithLValueExpressionPlaceholder(boundWithLValueExpressionPlaceholder As BoundWithLValueExpressionPlaceholder) As IInstanceReferenceOperation
+            Dim referenceKind As InstanceReferenceKind = InstanceReferenceKind.InitializedInstance
             Dim syntax As SyntaxNode = boundWithLValueExpressionPlaceholder.Syntax
             Dim type As ITypeSymbol = boundWithLValueExpressionPlaceholder.Type
             Dim constantValue As [Optional](Of Object) = ConvertToOptional(boundWithLValueExpressionPlaceholder.ConstantValueOpt)
             Dim isImplicit As Boolean = boundWithLValueExpressionPlaceholder.WasCompilerGenerated
-            Return New InstanceReferenceExpression(_semanticModel, syntax, type, constantValue, isImplicit)
+            Return New InstanceReferenceExpression(referenceKind, _semanticModel, syntax, type, constantValue, isImplicit)
         End Function
 
         Private Function CreateBoundWithRValueExpressionPlaceholder(boundWithRValueExpressionPlaceholder As BoundWithRValueExpressionPlaceholder) As IInstanceReferenceOperation
+            Dim referenceKind As InstanceReferenceKind = InstanceReferenceKind.InitializedInstance
```

Comment:
>InitializedInstance [](start = 79, length = 19)

I think this could be an implicit receiver in an With statement. We probably should have a special kind for this case and use the appropriate one here.